pkg/textparse/parse.go

```diff
@@ -123,14 +124,22 @@ func (p *Parser) Metric(l *labels.Labels) string {
 		c := p.l.offsets[i+2] - p.l.mstart
 		d := p.l.offsets[i+3] - p.l.mstart
 
-		*l = append(*l, labels.Label{Name: s[a:b], Value: s[c:d]})
+		*l = append(*l, labels.Label{Name: s[a:b], Value: replacer.Replace(s[c:d])})
```

Comment:
This allocates the entire string again every time, even if nothing needs to be replaced - whereas I think we're trying to avoid any unnecessary allocations here?

Perhaps we would need a custom replacer that only allocates a new slice when there are escape sequences?